src/engine.c

```diff
@@ -248,104 +248,90 @@ ibus_hangul_engine_get_type (void)
 void
 ibus_hangul_init (IBusBus *bus)
 {
-    GVariant* value;
+    GVariant* value = NULL;
 
     hanja_table = hanja_table_load (NULL);
 
     symbol_table = hanja_table_load (IBUSHANGUL_DATADIR "/data/symbol.txt");
 
-    config = ibus_bus_get_config (bus);
-    if (config)
-        g_object_ref_sink (config);
+    settings_hangul = g_settings_new ("org.freedesktop.ibus.engine.hangul");
+    settings_panel = g_settings_new ("org.freedesktop.ibus.panel");
 
-    hangul_keyboard = g_string_new_len ("2", 8);
-    value = ibus_config_get_value (config, "engine/hangul",
-                                         "hangul-keyboard");
+    hangul_keyboard = g_string_new_len (NULL, 8);
+    value = g_settings_get_value (settings_hangul, "hangul-keyboard");
     if (value != NULL) {
         const gchar* str = g_variant_get_string (value, NULL);
         g_string_assign (hangul_keyboard, str);
-        g_variant_unref(value);
+        g_clear_pointer (&value, g_variant_unref);
     }
 
     hotkey_list_init(&switch_keys);
 
-    value = ibus_config_get_value (config, "engine/hangul",
-                                         "switch-keys");
+    value = g_settings_get_value (settings_hangul, "switch-keys");
     if (value != NULL) {
         const gchar* str = g_variant_get_string (value, NULL);
         hotkey_list_set_from_string(&switch_keys, str);
-        g_variant_unref(value);
-    } else {
-	hotkey_list_append(&switch_keys, IBUS_Hangul, 0);
```

Comment:
I would prefer handling the unpredictable error here.
Switch keys are quite important.